src/signals/incident-management/containers/IncidentDetail/IncidentDetail.tsx

```diff
@@ -235,18 +227,10 @@ const IncidentDetail = () => {
     getIncident(`${configuration.INCIDENT_PRIVATE_ENDPOINT}${id}`)
   }, [getIncident, id])
 
-  const prevIncidentId = useRef<string>(id)
   const retrieveUnderlyingData = useCallback(() => {
```

Comment:
when calling retrieveUnderlying data just get this. All the if'ies make the call very complex.